### PR TITLE
Get spindle data

### PR DESCRIPTION
**Pull request by DennisKiselev**
Show get spindle data button when SC2 is in use, which can be used to show a popup with spindle info on it (not localised)

Tested on rig, both connected and disconnected spindle

image
image
image

### CODE REVIEW

**Comment by DennisKiselev** [comment]
image

**Comment by LettyYeti** [comment]
Recommend merging master back into both branches to make sure that the flat spindle image updates aren't overwritten by graphical changes in the this PR

File: src/asmcnc/apps/maintenance_app/widget_maintenance_spindle_settings.py [hunk 1]
```diff
@@ -377,49 +374,71 @@ def autofill_rpm_time(self):
...
+        if self.m.state().startswith('Idle'):
+            self.wait_popup = popup_info.PopupWait(self.sm, self.l, self.l.get_str('SmartBench is raising the Z axis.'))
+            self.m.zUp()
+            Clock.schedule_once(self.get_spindle_data, 0.4)
+        else:
+            popup_info.PopupError(self.sm, self.l, self.l.get_str("Please ensure machine is idle before continuing."))
+
+    def get_spindle_data(self, dt):
+        # Wait until Z axis is raised
+        if self.m.state().startswith('Idle'):
```
Comment (LettyYeti) [inline src/asmcnc/apps/maintenance_app/widget_maintenance_spindle_settings.py:404]: probably safer to use if not self.m.smartbench_is_busy() instead - it's a bit more thorough

**Review by LettyYeti — CHANGES_REQUESTED** [review]

**Comment by LettyYeti** [comment]
In the SC2 spindle data popup, hardly any info is reported. Does your spindle not have any info programmed into it?

**Comment by DennisKiselev** [comment]
> Recommend merging master back into both branches to make sure that the flat spindle image updates aren't overwritten by graphical changes in the this PR

Flat spindle images PR is off of this branch, so I will merge that one first before this one and there should be no conflicts

**Comment by DennisKiselev** [comment]
> In the SC2 spindle data popup, hardly any info is reported. Does your spindle not have any info programmed into it?

It's a screenshot from windows, so there won't be any real data, it definitely works correctly on the rig 🙂

**Comment by DennisKiselev** [comment]
image

**Pull request merged**